### ISSUE TITLE
Restarting container with secure password

### ISSUE THREAD

**Issue by Koubek** [Microsoft/nav-docker#91]
Houston, I think we have a problem...
When (re)starting an existing container that was initialized with the new secure password feature and the key was removed the following error (logically) appears:
![image](https://user-images.githubusercontent.com/24265973/32407841-ecba4278-c18e-11e7-8078-3dd639ee00a8.png)

We need to change `SetupVariables.ps1` script but to do so we would need to assign `$restartingInstance` a bit earlier, at least right before `SetupVariables.ps1` is being called. Ideally, I would set this variable at the as soon as possible. I would make changes in the scripts but I know your solution will be better... ;)

One possible way would be:
+ in `navstart.ps1`: move the following at the beginning of the script:
```PowerShell
$publicDnsNameFile = "$RunPath\PublicDnsName.txt"
$publicDnsNameChanged = $false
$restartingInstance = Test-Path -Path $publicDnsNameFile -PathType Leaf
```
+ in `SetupVariables.ps1` change the behavior to assign the secure password only in this case `$restartingInstance -eq $false`.

But then we can\`t use (reference) `$securePassword` later in this run (restart). This need to be reviewed.

**Comment by freddydk** [comment]
Thanks - yeah it sounds like you found a bad one there.
The workaround for now would be to add -e RemovePasswordKeyFile=N, but I will have a look at your suggestion and how to fix this.

**Comment by Koubek** [comment]
IMHO you could skip some of the scripts using `$securePassword` (eg: `SetupNavUsers.ps1` etc.) when restarting. This could make the restart even faster (not much)... The service, SQL and Windows account should be already preconfigured from the previous start. Am I right?

**Comment by freddydk** [comment]
That should already happen - but not enough

Få Outlook til iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Koubek** [comment]
You are right, I can see,

**Comment by Koubek** [comment]
I have tested the changes and and the container started fine.

**Comment by Koubek** [comment]
Also with **gMSA** the current version seems to be failing. Even worse, this happens during the `docker run` (as the variable does not exist).

```
Ready for connections!
Cannot find a variable with the name 'securePassword'.
```

My fix proposal #92 is fixing this issue as well.

**Comment by Koubek** [comment]
If you want to test it, you can copy the modified files into your `my` folder and that's it :) The changes affect only the runtime parts of the code so...

**Comment by freddydk** [comment]
This will be fixed in generic version 0.0.3.2 - and I will rebuild all images to be based on this

**Comment by Koubek** [comment]
OK, perfect!!! Thanks!
